File: backend/app/schemas/anime.py
from pydantic import BaseModel


class AnimeSearchResult(BaseModel):
    id: int
    slug: str
    title: str
    title_eng: str | None = None
    cover_url: str | None = None
    type: str | None = None
    year: str | None = None
    episodes_count: int | None = None
    genres: list[str] = []
    dub: bool = False


class AnimeDetail(BaseModel):
    id: int
    slug: str
    title: str
    title_eng: str | None = None
    cover_url: str | None = None
    banner_url: str | None = None
    plot: str | None = None
    type: str | None = None
    year: str | None = None
    episodes_count: int | None = None
    genres: list[str] = []
    status: str | None = None
    dub: bool = False


class Episode(BaseModel):
    id: int
    number: str
    title: str | None = None
    created_at: str | None = None
    views: int | None = None
    download_status: str | None = None


class EpisodesResponse(BaseModel):
    episodes: list[Episode]
    total: int
    has_more: bool


class SearchResponse(BaseModel):
    results: list[AnimeSearchResult]

File: frontend/src/types/anime.ts
export interface AnimeSearchResult {
  id: number;
  slug: string;
  title: string;
  title_eng: string | null;
  cover_url: string | null;
  type: string | null;
  year: string | null;
  episodes_count: number | null;
  genres: string[];
  dub: boolean;
  source_site: string;
}

export interface AnimeDetail {
  id: number;
  slug: string;
  title: string;
  title_eng: string | null;
  cover_url: string | null;
  banner_url: string | null;
  plot: string | null;
  type: string | null;
  year: string | null;
  episodes_count: number | null;
  genres: string[];
  status: string | null;
  dub: boolean;
  source_site: string;
}

export interface Episode {
  id: number;
  number: string;
  title: string | null;
  created_at: string | null;
  views: number | null;
  download_status: string | null;
}

export interface EpisodesResponse {
  episodes: Episode[];
  total: number;
  has_more: boolean;
}

export interface SearchResponse {
  results: AnimeSearchResult[];
}
